kernel/src/fs/utils/endpoint.rs

```diff
@@ -222,6 +224,22 @@ impl<T: AsRef<EndpointState>> Endpoint<T> {
             .notify(IoEvents::HUP | IoEvents::ERR | IoEvents::OUT);
     }
 
+    /// Activates the local endpoint.
+    pub fn activate(&self) {
```

Comment:
I'm not sure if it's really a good idea to have the `activate()` method. But I think the current usage is perhaps fine. Let's restrict the visibility to the current file or the `fs` module.